pylint/checkers/classes/class_checker.py

```diff
@@ -1586,8 +1594,12 @@ def _check_slots(self, node: nodes.ClassDef) -> None:
 
     def _get_classdef_slots_names(self, node: nodes.ClassDef) -> list[str]:
 
-        slots_names = []
-        for slots in node.ilookup("__slots__"):
+        slots_names: list[str] = []
+        try:
+            inferred_slots = tuple(node.ilookup("__slots__"))
+        except astroid.InferenceError:  # pragma: no cover
```

Comment:
This is no-covered because the calls to this function are (currently) guarded by `_has_valid_slots()`, which does the same test, but just adding safety because we should be failing to catch InferenceError from calls to the inference system.